### PR TITLE
Add ten-color accessible color cycle as style sheet

### PR DESCRIPTION
**Pull request by mpetroff**
## PR summary

(Following up from email.)

Adds a style sheet with the ten-color color cycle survey palette from Petroff (2021) [which will be published in a journal eventually]:
https://arxiv.org/abs/2107.02270
https://github.com/mpetroff/accessible-color-cycles

Related to (but does not fix) #9460.

At the moment, this PR contains only the minimal changes. I've only added the ten-color sequence from the paper and only done so as a style sheet.

The paper additionally contains six- and eight-color sequences, which could also be added.

Would it be better to add to `_BUILTIN_COLOR_SEQUENCES` in `lib/matplotlib/colors.py` with data in `lib/matplotlib/_cm.py` instead of as a style sheet?

I'm also open to naming suggestions. The current name is the placeholder I've been using, with `ccs` standing for "color cycle survey".

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [x] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by tacaswell** [comment]
If it is ready in the next week or so, it can go in 3.9, but I think it is important to get the names and such rather rather than rushing it through.

**Comment by kpedro88** [comment]
@mpetroff great to see this! As a user, I am in favor of adding the 6- and 8-color cycles.

File: doc/users/next_whats_new/ccs_color_cycle.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+New more-accessible color cycle
+-------------------------------
+
+A new color cycle named 'ccs10' was added. This cycle was constructed using a
```
Comment (matthewfeickert) [inline doc/users/next_whats_new/ccs_color_cycle.rst:4]:
```suggestion
A new color cycle named 'petroff10' was added. This cycle was constructed using a
```

File: doc/users/next_whats_new/ccs_color_cycle.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+developed from a crowdsourced color-preference survey. It aims to be both
+generally pleasing aesthetically and colorblind accessible such that it could
+serve as a default in the aim of universal design. For more details
+see `Petroff, M. A.: "Accessible Color Sequences for Data Visualization"
+<https://arxiv.org/abs/2107.02270>`_ and related `SciPy talk`_. A demonstration
+is included in the style sheets reference_. To load this color cycle in place
+of the default::
+
+  import matplotlib.pyplot as plt
+  plt.style.use('ccs10')
```
Comment (matthewfeickert) [inline doc/users/next_whats_new/ccs_color_cycle.rst:16]:
```suggestion
  plt.style.use('petroff10')
```

File: lib/matplotlib/mpl-data/stylelib/ccs10.mplstyle
Comment (matthewfeickert) [inline lib/matplotlib/mpl-data/stylelib/ccs10.mplstyle:1]: This would get renamed to `petroff10.mplstyle`.

**Review by matthewfeickert — CHANGES_REQUESTED** [review]
Given the discussion in https://github.com/mpetroff/accessible-color-cycles/issues/1 this PR would adopt the `petroffX` name scheme.

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
I took the liberty of accepting the naming change, adding it to the sequence registry, and fixing the rebase.

**Comment by matthewfeickert** [comment]
@mpetroff can we get an approval from you that you're good with the name changes discussed in https://github.com/mpetroff/accessible-color-cycles/issues/1 that were adopted here?

**Comment by mpetroff** [comment]
> @mpetroff can we get an approval from you that you're good with the name changes discussed in https://github.com/mpetroff/accessible-color-cycles/issues/1 that were adopted here?

Yes, I'm good with it.

**Comment by matthewfeickert** [comment]
@tacaswell @mpetroff Can we rebase this off `main` again as the CI seems to be fixed there now, and then this should be good to merge whenever.

**Comment by tacaswell** [comment]
I went ahead and merged this and opened issue for the additional examples / documentation.

**Comment by tacaswell** [comment]
Thank you @mpetroff !

**Comment by matthewfeickert** [comment]
Probably the wrong place to ask this, but `petroff10` landed in Matplotlib `v3.10.0`

https://matplotlib.org/stable/users/prev_whats_new/whats_new_3.10.0.html#new-more-accessible-color-cycle

so we can now do

```python
import matplotlib.pyplot as plt
plt.style.use("petroff10")

petroff10
```

which is great. :rocket: (Also nice [`petroff10` style sheet example](https://matplotlib.org/stable/gallery/style_sheets/petroff10.html#petroff10-style-sheet)!)

For some high energy physics training sessions the 6 color scheme is sometimes demoed (c.f. the [`mplhep` example from the 2024 USCMS/IRIS-HEP Analysis Software Training](https://github.com/iris-hep/uscms-idap-training/blob/2998688069ee282174520b221e368f92e9ab840a/mplhep/hist_mplhep_basics.ipynb)). To do a 6 color cycle should one just truncate the 10 color cycle?

**Comment by timhoffm** [comment]
The 6 color sequence by Petroff is not just a truncated 10 color sequence. It has distinct colors. You can read thrombus here and make a custom color sequence https://arxiv.org/abs/2107.02270.

**Pull request merged**